robot/client/client.go

```diff
@@ -680,10 +688,10 @@ func (rc *RobotClient) ResourceByName(name resource.Name) (resource.Resource, er
 func (rc *RobotClient) createClient(name resource.Name) (resource.Resource, error) {
 	apiInfo, ok := resource.LookupGenericAPIRegistration(name.API)
 	if !ok || apiInfo.RPCClient == nil {
-		return grpc.NewForeignResource(name, &rc.conn), nil
+		return grpc.NewForeignResource(name, rc.getClientConn()), nil
```

Comment:
There's a comment at the top of the `getClientConn` method that says: `// Must be called with `rc.mu` in ReadLock+ mode.` Are we doing that here? I assume the lock is already before we call `createClient`?

Reply:
Yep -- need to unfold up a few lines: https://github.com/viamrobotics/rdk/pull/4751/files#diff-d6a5d93563cdb968513a4bc0246d93a5568b7233c639bfd2ba3e1d42c3925083R667